libdebug/ptrace/ptrace_interface.py

```diff
@@ -350,6 +350,55 @@ def finish(self: PtraceInterface, thread: ThreadContext, heuristic: str) -> None
         else:
             raise ValueError(f"Unimplemented heuristic {heuristic}")
 
+    def next(self: PtraceInterface, thread: ThreadContext) -> None:
+        """Executes the next instruction of the process. If the instruction is a call, the debugger will continue until the function returns.
+        """
+
+        # Check if the current instruction is a call and its skip
+        is_call = self.lib_trace.is_call_instruction(
+            self._global_state,
+            thread.thread_id
+        )
+
+        if is_call:
+            skip_address = self.lib_trace.compute_call_skip(
+                self._global_state,
+                thread.thread_id
+            )
+        
+        # Step forward
+        self.step(thread)
```

Comment:
I have some doubts about the logic here: is `is_call` is True, do we need to step?

Reply:
Actually, you are correct. This is a piece of logic that I had structured for the first implementation attempt that used the backtrace (like the finish). I will correct it.

Reply:
Resolved in commit [9e855f6](https://github.com/libdebug/libdebug/commit/9e855f6e3ddf478fadbb735e8701f220cea3f867)